File: FbxCppTests/FbxDataTypeTest.cpp

#include "Tests.h"

using namespace std;

void FbxDataType_DefaultConstructor_AttributesSet()
{
    // when:
    FbxDataType dt = FbxDataType::Create("int", EFbxType::eFbxInt);
    // then:
    AssertTrue(dt.Valid());
    AssertEqual("int", dt.GetName());
    AssertEqual(EFbxType::eFbxInt, dt.GetType());
}

void FbxDataType_OperatorEquals_MatchesSelfButNotIdenticalObjects()
{
    // when:
    FbxDataType dt1 = FbxDataType::Create("int", EFbxType::eFbxInt);
    FbxDataType dt2 = FbxDataType::Create("int", EFbxType::eFbxInt);

    // then:
    AssertTrue(dt1.Valid());
    AssertTrue(dt2.Valid());
    AssertEqual(dt1.GetName(), dt2.GetName());
    AssertEqual(dt1.GetType(), dt2.GetType());
    AssertEqual(dt1, dt1);
    AssertEqual(dt1, dt1);
    AssertTrue(dt1 == dt1);
    AssertTrue(dt2 == dt2);
    AssertFalse(dt1 != dt1);
    AssertFalse(dt2 != dt2);
    AssertNotEqual(FbxIntDT, dt1);
    AssertNotEqual(FbxIntDT, dt2);
    AssertFalse(FbxIntDT == dt1);
    AssertFalse(FbxIntDT == dt2);
    AssertTrue(FbxIntDT != dt1);
    AssertTrue(FbxIntDT != dt2);
    AssertNotEqual(dt1, dt2);
    AssertFalse(dt1 == dt2);
    AssertTrue(dt1 != dt2);
}

void FbxDataType_FbxGetDataTypeFromEnum_YieldsCorrectDataType()
{
    // expect:
    AssertEqual(FbxUndefinedDT, FbxGetDataTypeFromEnum(EFbxType::eFbxUndefined));
    AssertEqual(FbxCharDT, FbxGetDataTypeFromEnum(EFbxType::eFbxChar));
    AssertEqual(FbxUCharDT, FbxGetDataTypeFromEnum(EFbxType::eFbxUChar));
    AssertEqual(FbxShortDT, FbxGetDataTypeFromEnum(EFbxType::eFbxShort));
    AssertEqual(FbxUShortDT, FbxGetDataTypeFromEnum(EFbxType::eFbxUShort));
    AssertEqual(FbxUIntDT, FbxGetDataTypeFromEnum(EFbxType::eFbxUInt));
    AssertEqual(FbxLongLongDT, FbxGetDataTypeFromEnum(EFbxType::eFbxLongLong));
    AssertEqual(FbxULongLongDT, FbxGetDataTypeFromEnum(EFbxType::eFbxULongLong));
    AssertEqual(FbxHalfFloatDT, FbxGetDataTypeFromEnum(EFbxType::eFbxHalfFloat));
    AssertEqual(FbxBoolDT, FbxGetDataTypeFromEnum(EFbxType::eFbxBool));
    AssertEqual(FbxIntDT, FbxGetDataTypeFromEnum(EFbxType::eFbxInt));
    AssertEqual(FbxFloatDT, FbxGetDataTypeFromEnum(EFbxType::eFbxFloat));
    AssertEqual(FbxDoubleDT, FbxGetDataTypeFromEnum(EFbxType::eFbxDouble));
    AssertEqual(FbxDouble2DT, FbxGetDataTypeFromEnum(EFbxType::eFbxDouble2));
    AssertEqual(FbxDouble3DT, FbxGetDataTypeFromEnum(EFbxType::eFbxDouble3));
    AssertEqual(FbxDouble4DT, FbxGetDataTypeFromEnum(EFbxType::eFbxDouble4));
    AssertEqual(FbxDouble4x4DT, FbxGetDataTypeFromEnum(EFbxType::eFbxDouble4x4));
    AssertEqual(FbxEnumDT, FbxGetDataTypeFromEnum(EFbxType::eFbxEnum));
    AssertEqual(FbxEnumDT, FbxGetDataTypeFromEnum(EFbxType::eFbxEnumM));
    AssertEqual(FbxStringDT, FbxGetDataTypeFromEnum(EFbxType::eFbxString));
    AssertEqual(FbxTimeDT, FbxGetDataTypeFromEnum(EFbxType::eFbxTime));
    AssertEqual(FbxReferenceDT, FbxGetDataTypeFromEnum(EFbxType::eFbxReference));
    AssertEqual(FbxBlobDT, FbxGetDataTypeFromEnum(EFbxType::eFbxBlob));
    AssertEqual(FbxDistanceDT, FbxGetDataTypeFromEnum(EFbxType::eFbxDistance));
    AssertEqual(FbxDateTimeDT, FbxGetDataTypeFromEnum(EFbxType::eFbxDateTime));
}

void FbxDataTypeTest::RegisterTestCases()
{
    AddTestCase(FbxDataType_DefaultConstructor_AttributesSet);
    AddTestCase(FbxDataType_OperatorEquals_MatchesSelfButNotIdenticalObjects);
    AddTestCase(FbxDataType_FbxGetDataTypeFromEnum_YieldsCorrectDataType);
}

File: FbxSharpTests/FbxDataTypeTest.cs
using System;
using NUnit.Framework;
using FbxSharp;

namespace FbxSharpTests
{
    [TestFixture]
    public class FbxDataTypeTest : TestBase
    {
        [Test]
        public void FbxDataType_DefaultConstructor_AttributesSet()
        {
            // when:
            var dt = FbxDataType.Create("int", EFbxType.eFbxInt);
            // then:
            Assert.True(dt.Valid());
            Assert.That(dt.GetName(), Is.EqualTo("int"));
            Assert.That(dt.GetFbxType(), Is.EqualTo(EFbxType.eFbxInt));
        }

        [Test]
        public void FbxDataType_OperatorEquals_MatchesSelfButNotIdenticalObjects()
        {
            // when:
            var dt1 = FbxDataType.Create("int", EFbxType.eFbxInt);
            var dt2 = FbxDataType.Create("int", EFbxType.eFbxInt);

            // then:
            Assert.True(dt1.Valid());
            Assert.True(dt2.Valid());
            Assert.That(dt2.GetName(), Is.EqualTo(dt1.GetName()));
            Assert.That(dt2.GetFbxType(), Is.EqualTo(dt1.GetFbxType()));
            Assert.That(dt1, Is.EqualTo(dt1));
            Assert.That(dt1, Is.EqualTo(dt1));
            Assert.True(dt1 == dt1);
            Assert.True(dt2 == dt2);
            Assert.False(dt1 != dt1);
            Assert.False(dt2 != dt2);
            Assert.That(dt1, Is.Not.EqualTo(FbxDataTypes.FbxIntDT));
            Assert.That(dt2, Is.Not.EqualTo(FbxDataTypes.FbxIntDT));
            Assert.False(FbxDataTypes.FbxIntDT == dt1);
            Assert.False(FbxDataTypes.FbxIntDT == dt2);
            Assert.True(FbxDataTypes.FbxIntDT != dt1);
            Assert.True(FbxDataTypes.FbxIntDT != dt2);
            Assert.That(dt2, Is.Not.EqualTo(dt1));
            Assert.False(dt1 == dt2);
            Assert.True(dt1 != dt2);
        }

        [Test]
        public void FbxDataType_FbxGetDataTypeFromEnum_YieldsCorrectDataType()
        {
            // expect:
            Assert.That(FbxDataType.FbxGetDataTypeFromEnum(EFbxType.eFbxUndefined), Is.EqualTo(FbxDataTypes.FbxUndefinedDT));
            Assert.That(FbxDataType.FbxGetDataTypeFromEnum(EFbxType.eFbxChar), Is.EqualTo(FbxDataTypes.FbxCharDT));
            Assert.That(FbxDataType.FbxGetDataTypeFromEnum(EFbxType.eFbxUChar), Is.EqualTo(FbxDataTypes.FbxUCharDT));
            Assert.That(FbxDataType.FbxGetDataTypeFromEnum(EFbxType.eFbxShort), Is.EqualTo(FbxDataTypes.FbxShortDT));
            Assert.That(FbxDataType.FbxGetDataTypeFromEnum(EFbxType.eFbxUShort), Is.EqualTo(FbxDataTypes.FbxUShortDT));
            Assert.That(FbxDataType.FbxGetDataTypeFromEnum(EFbxType.eFbxUInt), Is.EqualTo(FbxDataTypes.FbxUIntDT));
            Assert.That(FbxDataType.FbxGetDataTypeFromEnum(EFbxType.eFbxLongLong), Is.EqualTo(FbxDataTypes.FbxLongLongDT));
            Assert.That(FbxDataType.FbxGetDataTypeFromEnum(EFbxType.eFbxULongLong), Is.EqualTo(FbxDataTypes.FbxULongLongDT));
            Assert.That(FbxDataType.FbxGetDataTypeFromEnum(EFbxType.eFbxHalfFloat), Is.EqualTo(FbxDataTypes.FbxHalfFloatDT));
            Assert.That(FbxDataType.FbxGetDataTypeFromEnum(EFbxType.eFbxBool), Is.EqualTo(FbxDataTypes.FbxBoolDT));
            Assert.That(FbxDataType.FbxGetDataTypeFromEnum(EFbxType.eFbxInt), Is.EqualTo(FbxDataTypes.FbxIntDT));
            Assert.That(FbxDataType.FbxGetDataTypeFromEnum(EFbxType.eFbxFloat), Is.EqualTo(FbxDataTypes.FbxFloatDT));
            Assert.That(FbxDataType.FbxGetDataTypeFromEnum(EFbxType.eFbxDouble), Is.EqualTo(FbxDataTypes.FbxDoubleDT));
            Assert.That(FbxDataType.FbxGetDataTypeFromEnum(EFbxType.eFbxDouble2), Is.EqualTo(FbxDataTypes.FbxDouble2DT));
            Assert.That(FbxDataType.FbxGetDataTypeFromEnum(EFbxType.eFbxDouble3), Is.EqualTo(FbxDataTypes.FbxDouble3DT));
            Assert.That(FbxDataType.FbxGetDataTypeFromEnum(EFbxType.eFbxDouble4), Is.EqualTo(FbxDataTypes.FbxDouble4DT));
            Assert.That(FbxDataType.FbxGetDataTypeFromEnum(EFbxType.eFbxDouble4x4), Is.EqualTo(FbxDataTypes.FbxDouble4x4DT));
            Assert.That(FbxDataType.FbxGetDataTypeFromEnum(EFbxType.eFbxEnum), Is.EqualTo(FbxDataTypes.FbxEnumDT));
            Assert.That(FbxDataType.FbxGetDataTypeFromEnum(EFbxType.eFbxEnumM), Is.EqualTo(FbxDataTypes.FbxEnumDT));
            Assert.That(FbxDataType.FbxGetDataTypeFromEnum(EFbxType.eFbxString), Is.EqualTo(FbxDataTypes.FbxStringDT));
            Assert.That(FbxDataType.FbxGetDataTypeFromEnum(EFbxType.eFbxTime), Is.EqualTo(FbxDataTypes.FbxTimeDT));
            Assert.That(FbxDataType.FbxGetDataTypeFromEnum(EFbxType.eFbxReference), Is.EqualTo(FbxDataTypes.FbxReferenceDT));
            Assert.That(FbxDataType.FbxGetDataTypeFromEnum(EFbxType.eFbxBlob), Is.EqualTo(FbxDataTypes.FbxBlobDT));
            Assert.That(FbxDataType.FbxGetDataTypeFromEnum(EFbxType.eFbxDistance), Is.EqualTo(FbxDataTypes.FbxDistanceDT));
            Assert.That(FbxDataType.FbxGetDataTypeFromEnum(EFbxType.eFbxDateTime), Is.EqualTo(FbxDataTypes.FbxDateTimeDT));
        }
    }
}
